server/lib/models/topic.go

```diff
@@ -10,6 +18,147 @@ type Topic struct {
 }
 
 func InitTopic() {
-	create_table(Topic{}, "topics", "Id")
+	t := create_table(Topic{}, "topics", "Id")
+	t.AddIndex("locale", "INDEX", []string{"Locale"})
+}
+
+const TOPIC_FETCH_LIMIT = 50
+
+func NewTopic(dbif Dbif, a *Account, req *proto.TopicCreateRequest) (*Topic, error) {
+	var (
+		p *Persona
+		c *Channel
+		err error
+	)
+	if p, err = a.Persona(dbif, req.Channel); err != nil {
+		return nil, &proto.Err{ Type: proto.Error_TopicCreate_NoPersonaError }
+	}
+	if c, err = FindChannel(dbif, req.Channel); err != nil {
+		return nil, &proto.Err{ Type: proto.Error_TopicCreate_DatabaseError }
+	}
+	body := proto.Model_Topic_Body {
+		ChannelName: c.Name,
+		Name: p.Name,
+	}
+	bs, err := body.Marshal()
+	if err != nil {
+		return nil, &proto.Err{ Type: proto.Error_TopicCreate_DatabaseError }
+	}
+	tp := &Topic {
+		proto.Model_Topic {
+			Id: dbm.UUID(),
+			Channel: req.Channel, 
+			Persona: p.Id,
+			Body: bs,
+		},
+	}
+	if err = dbif.Insert(tp); err != nil {
+		return nil, &proto.Err{ Type: proto.Error_TopicCreate_DatabaseError }
+	}
+	return tp, nil
+}
+
+//return value of actor call
+type HotEntry struct {
+	Id proto.UUID
```

Comment:
__[gometalinter]__
* [golint] struct field Id should be ID
* [gotype] undeclared name: proto